### PR TITLE
fix(curriculum): revise some of the tests for recipe lab

### LINKED ISSUE #56860: [ Full Stack Cert ] - update tests for recipe lab
The following  test passes and doesn’t actually check for text inside the paragraph element 

You should have a `p` element below your `h1` element with text describing your recipe.

```js
assert.strictEqual(document.querySelector('p')?.previousElementSibling.tagName, 'H1');
```

Here is the current passing code

```html
<body>
  <h1>some random title</h1>
  <p></p>
</body>
```

It will probably be better to split that test into two tests

One test to check that it is actually below the h1
Once test to check that there is some sort of text between the paragraph tags

Here is the file you need to edit

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md

### PR DESCRIPTION
**Pull request by CracktheDom**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56860

### CODE REVIEW

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
The issue request that the test is split into two

also please use [specific assert methods](https://www.chaijs.com/api/assert/) and [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) where needed.

File: curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md [hunk 1]
```diff
@@ -89,7 +89,15 @@ assert.lengthOf(document.querySelectorAll('h1'), 1);
...
-assert.strictEqual(document.querySelector('p')?.previousElementSibling.tagName, 'H1');
+// 1. Check that a <p> tag is below the <h1> tag
+const h1 = document.querySelector('h1');
+const p = document.querySelector('p');
+assert(h1 && p && h1.compareDocumentPosition(p) 
+  && Node.DOCUMENT_POSITION_FOLLOWING
+);
+
+// 2. Check that the <p> tag has text content
+assert(p && p.textContent.trim().length > 0,);
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md:100]:
````suggestion
Your first `p` element should describe your recipe.

```js
assert.isNotEmpty(document.querySelector('p')?.textContent?.trim());
````

File: curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md [hunk 2]
```diff
@@ -89,7 +89,15 @@ assert.lengthOf(document.querySelectorAll('h1'), 1);
 You should have a `p` element below your `h1` element with text describing your recipe.
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md:89]:
```suggestion
You should have a `p` element below your `h1` element.
```

File: curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md [hunk 3]
```diff
@@ -89,7 +89,15 @@ assert.lengthOf(document.querySelectorAll('h1'), 1);
...
 
 ```js
-assert.strictEqual(document.querySelector('p')?.previousElementSibling.tagName, 'H1');
+// 1. Check that a <p> tag is below the <h1> tag
+const h1 = document.querySelector('h1');
+const p = document.querySelector('p');
+assert(h1 && p && h1.compareDocumentPosition(p) 
+  && Node.DOCUMENT_POSITION_FOLLOWING
+);
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md:97]:
````suggestion
assert.strictEqual(document.querySelector('h1')?.nextElementSibling, document.querySelector('p'));
```
````

**Comment by ilenia-magoni** [comment]
If you go to the files tab you can add all the suggestions to a abtch and make a single commit

**Review by CracktheDom — COMMENTED** [review]
Add suggestions to batch

**Review by CracktheDom — COMMENTED** [review]
Add suggestions

File: curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md (hunk 2 above)
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md:89]: Why did you not accept this @CracktheDom ?

**Review by majestic-owl448 — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Pull request merged**